apps/src/code-studio/responsive.js

```diff
@@ -0,0 +1,15 @@
+import debounce from 'lodash/debounce';
+import { getStore } from './redux';
+import { setResponsiveSize } from './responsiveRedux';
+
+/**
+ * Listen for page resize and dispatch events to Redux when if cross a
+ * responsive breakpoint width.
+ */
+export default function initResponsive() {
+  const store = getStore();
+
+  window.addEventListener('resize', debounce(() => {
+    store.dispatch(setResponsiveSize());
```

Comment:
This should be providing a param, right? Should we actually be providing the window width vs. the responsiveSize breakpoint? (possible im misunderstanding something)

Reply:
Ah, right, I see what you're saying.  I'll move the logic out of the reducer and into this file.

For some reason I thought I should provide initial state via `getResponsiveBreakpoint` and moved the function definition to the reducer.  Instead, I think `response.js` calls `setResponsiveSize()` immediately with the initial size category (lg/md/sm/xs) in addition to setting up the event listener.  Does that sound right?

Reply:
I think I was maybe thinking about this all wrong. Let me start fresh and state some assumptions about what we're trying to accomplish to see if they are right:

- Goal: We want to be able to easily tell how big the current window is, but bucketed into a few different categories - i.e. we want to know if it is sx, sm, md, or lg.

If true, this is what I would do:

1. Provide a setWindowSizeAction
2. Have an eventListener that dispatches `setWindowSizeAction` every time we resize (possibly/probably debounced.
3. Have a reduce that listens for the setWindowSize action, and returns the breakpoint (xs, sm, etc.) for that size.

Reply:
That makes sense!  Any reason to have the conversion of width in px to lg/md/sm/xs live in the reducer vs. in `responsive.js`?

I guess what I'm asking is whether `setWindowSizeAction` should take a px width of the exact window size or one of lg/md/sm/xs.  Thanks!

Reply:
Alternatively, we could store the raw window size, and provide a helper so that instead of something like
```
connect(state => {
  breakpoint: state.responsiveBreakpoint
})
```
we would do
```
connect(state => {
  breakpoint: responsiveBreakpoint(state.windowWidth)
}
```
Just storing the breakpoint is maybe easier to consume.

Reply:
I think either way works (passing in breakpoint or size)

Reply:
Having the conversion happen in the reducer perhaps make it easier to test.